cuqi/likelihood/_likelihood.py

```diff
@@ -91,12 +107,19 @@ def model(self) -> Model:
         
         return model_value
 
-    def __call__(self, *args, **kwargs):
+    def _condition(self, *args, **kwargs):
```

Comment:
One way to think of it is that __call__ is a short-hand for the _condition method. Currently the _condition method is not exposed to the user because we recommend using __call__. But it is more clear in the code base naming the method _conditon.

I think giving Likelihood and EvaluatedDensity their own implementations of _condition is better for the codebase. This is because they are inherently different objects that have different behavior when conditioning.

The code for conditioning on Likelihood and EvaluatedDensity is quite simple however. So maybe not a huge deal?